public/app/plugins/datasource/cloud-monitoring/datasource.ts

```diff
@@ -350,14 +350,14 @@ export default class CloudMonitoringDatasource extends DataSourceWithBackend<
   }
 
   interpolateFilters(filters: string[], scopedVars: ScopedVars) {
-    const completeFilter: Filter[] = chunk(filters, 4)
+    const completeFilter = chunk(filters, 4)
       .map(([key, operator, value, condition]) => ({
         key,
         operator,
         value,
         ...(condition && { condition }),
       }))
-      .reduce((res, filter) => (filter.value ? [...res, filter] : res), []);
+      .reduce((res, filter) => (filter.value ? [...res, filter] : res), [] as Filter[]);
```

Comment:
imho `as` should be a strict error also (or a lint error). 

I'm trying to understand what this `.reduce()` actually does - it looks like it's a filter? I know you said you wanted to change only types, not code, but could the difficult-to-read reduce be replaced with... `.filter(filter => filter.value) ?`

Reply:
great catch - it definitely would be better as a filter 🙌